azure-pipelines.yml

```diff
@@ -138,13 +146,34 @@ stages:
               verbosityRestore: 'normal'  # Default is noisy 'Detailed'
 
           - task: DotNetCoreCLI@2
-            displayName: Dotnet generate SBOM
-            # https://sonarsource.atlassian.net/browse/BUILD-1303
+            displayName: Install CycloneDX
+            env:
+              ARTIFACTORY_USER: $(ARTIFACTORY_PRIVATE_READER_USERNAME)
+              ARTIFACTORY_PASSWORD: $(ARTIFACTORY_PRIVATE_READER_ACCESS_TOKEN)
             inputs:
               command: custom
-              custom: CycloneDX
-              projects: '$(SOLUTION)'
-              arguments: '-t -j -o build'
+              custom: tool
+              feedsToUse: 'select'
+              includeNuGetOrg: true
+              arguments: 'install --global CycloneDX'
+
+          - powershell: |
+              # It's not possible to use the tool using only its name (dotnet CycloneDX [args]), because of this message in the pipeline:
+              # "Since you just installed the .NET Core SDK, you will need to reopen the Command Prompt window before running the tool you installed."
+              # Instead, we need to find the installed tool and execute it directly using its absolute path.
+
+              # Find the latest tool version with the latest .NET runtime version
+              $toolVersionRegex = "^\d+(\.\d+)+$"
+              $netVersionRegex = "^net\d+(\.\d+)*$"
+
+              $latestToolVersionPath = Get-ChildItem -Path "$env:userprofile\.dotnet\tools\.store\cyclonedx" -Directory | Where-Object { $_.Name -match $toolVersionRegex } | Sort-Object Name -Descending | Select-Object -First 1
```

Comment:
The sorting could fail if the version gets high enough for double-digits. Use the version type instead: https://manualbashing.github.io/posts/sort-version-strings/

Reply:
Why do we even need to sort? Is there really more than one folder?

Reply:
No, you're right, it's only needed for the netX.Y folders.